Sources/Logging/Logging.swift

```diff
@@ -538,27 +572,9 @@ internal struct StdoutLogHandler: LogHandler {
         let prettyMetadata = metadata?.isEmpty ?? true
             ? self.prettyMetadata
             : self.prettify(self.metadata.merging(metadata!, uniquingKeysWith: { _, new in new }))
-        print("\(self.timestamp()) \(level):\(prettyMetadata.map { " \($0)" } ?? "") \(message)")
-    }
 
-    public var metadata: Logger.Metadata {
-        get {
-            return self.lock.withLock { self._metadata }
-        }
-        set {
-            self.lock.withLock { self._metadata = newValue }
-        }
-    }
-
-    public subscript(metadataKey metadataKey: String) -> Logger.Metadata.Value? {
-        get {
-            return self.lock.withLock { self._metadata[metadataKey] }
-        }
-        set {
-            self.lock.withLock {
-                self._metadata[metadataKey] = newValue
-            }
-        }
+        var stream = self.stream
+        stream.write("\(timestamp()) \(level):\(prettyMetadata.map { " \($0)" } ?? "") \(message)\n")
```

Comment:
@weissi I found that `Swift.print` actually calls `TextOutpuStream.write` multiple times. Looking at the stdlib implementation, they're able to ensure the underlying operations are performed with a single locking action (see https://github.com/apple/swift/blob/master/stdlib/public/core/Print.swift#L224-L231)

To ensure our lock works, I've switched to calling `write` directly. I've appended a newline to the message. Any fundamental issues you might see with that?

Reply:
@allenhumphreys that looks good to me! Thank you